internal/network/command_private_link_attachment_connection_create.go

```diff
@@ -0,0 +1,102 @@
+package network
+
+import (
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	networkingprivatelinkv1 "github.com/confluentinc/ccloud-sdk-go-v2/networking-privatelink/v1"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+)
+
+func (c *command) newPrivateLinkAttachmentConnectionCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create <name>",
+		Short: "Create a private link attachment connection.",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.privateLinkAttachmentConnectionCreate,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Create an AWS PrivateLink attachment connection.",
+				Code: "confluent network private-link attachment connection create aws-private-link-attachment-connection --cloud aws --endpoint vpce-1234567890abcdef0 --attachment platt-123456",
+			},
+		),
+	}
+
+	pcmd.AddCloudFlag(cmd)
+	cmd.Flags().String("endpoint", "", "ID of an AWS VPC endpoint that is connected to the AWS VPC endpoint service.")
+	c.addPrivateLinkAttachmentFlag(cmd)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	cobra.CheckErr(cmd.MarkFlagRequired("cloud"))
+	cobra.CheckErr(cmd.MarkFlagRequired("endpoint"))
+	cobra.CheckErr(cmd.MarkFlagRequired("attachment"))
+
+	return cmd
+}
+
+func (c *command) privateLinkAttachmentConnectionCreate(cmd *cobra.Command, args []string) error {
+	cloud, err := cmd.Flags().GetString("cloud")
+	if err != nil {
+		return err
+	}
+	cloud = strings.ToUpper(cloud)
+
+	endpoint, err := cmd.Flags().GetString("endpoint")
+	if err != nil {
+		return err
+	}
+
+	attachment, err := cmd.Flags().GetString("attachment")
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	createPrivateLinkAttachmentConnection := networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnection{
+		Spec: &networkingprivatelinkv1.NetworkingV1PrivateLinkAttachmentConnectionSpec{
+			DisplayName:           networkingprivatelinkv1.PtrString(args[0]),
+			Environment:           &networkingprivatelinkv1.ObjectReference{Id: environmentId},
+			PrivateLinkAttachment: &networkingprivatelinkv1.ObjectReference{Id: attachment},
+		},
+	}
+
+	switch cloud {
```

Comment:
is the value of `cloud` validated? If it's some invalid value, will it return an error?

Reply:
It's validated at the api level as `spec.cloud` will be missing if the value of `cloud` is not something that we expect. This is demonstrated and tested in https://github.com/confluentinc/cli/blob/ef124afee2ec0894cd6c97abb15a01fe10b14bc4/test/test-server/networking_handlers.go#L1171 and https://github.com/confluentinc/cli/blob/ef124afee2ec0894cd6c97abb15a01fe10b14bc4/test/network_test.go#L523